### PR TITLE
Document criteria for proposing new licenses

### PR DESCRIPTION
**Pull request by benbalter**
This pull request documents @mlinksva's suggestions for new license criteria over in https://github.com/github/choosealicense.com/issues/315.  Specifically:

> 1. The license must have [an SPDX identifier](https://spdx.org/licenses/). If you're license isn't register with SPDX, please [request that it be added](https://spdx.org/spdx-license-list/request-new-license-or-exception).
> 2. The license must be listed on one of the following approved lists of licenses:
>    - [List of OSI approved licenses](https://opensource.org/licenses/alphabetical)
>    - [GNU's list of free licenses](https://www.gnu.org/licenses/license-list.en.html) (_note: the license must be listed in one of three "free" categories_)
>    - [Open Definition's list of conformant licenses](http://opendefinition.org/licenses/) (non-code)
> 3. A [GitHub code search](https://github.com/search?q=MIT+filename%3ALICENSE&type=Code) must reveal at least _1,000_ public repositories using the license

If there's rough consensus that the above is good criteria, I'll adapt the existing tests to confirm the above (and run them against the existing license list). I believe SPDXiness is already checked, as is OSI approval, with a few whitelisted exceptions.

### CODE REVIEW

**Comment by benbalter** [comment]
Believe this is ready for :eyes:, especially from @mlinksva.

I added tests to enforce the proposed license addition criteria. All our existing licenses meet the criteria.

Excluding GitHub search results (which can't be retrieved programmatically), there are currently 104 potentially "approved" licenses:

```
aal, afl-1.1, afl-1.2, afl-2.0, afl-2.1, afl-3.0, agpl-3.0, apache-1.1, apache-2.0, apl-1.0, apsl-1.0, apsl-1.1, apsl-1.2, apsl-2.0, artistic-1.0, artistic-1.0-cl8, artistic-1.0-perl, artistic-2.0, bsd-2-clause, bsd-3-clause, bsl-1.0, catosl-1.1, cc-by-4.0, cc-by-sa-4.0, cc0-1.0, cddl-1.0, cecill-2.1, cnri-python, cpal-1.0, cpl-1.0, cua-opl-1.0, ecl-1.0, ecl-2.0, efl-1.0, efl-2.0, entessa, epl-1.0, eudatagrid, eupl-1.1, fair, frameworx-1.0, gpl-2.0, gpl-3.0, hpnd, ijg, imatix, intel, ipa, ipl-1.0, isc, lgpl-2.0, lgpl-2.1, lgpl-3.0, lpl-1.0, lpl-1.02, lppl-1.3c, miros, mit, motosoto, mpl-1.0, mpl-1.1, mpl-2.0, mpl-2.0-no-copyleft-exception, ms-pl, ms-rl, multics, nasa-1.3, naumen, ncsa, ngpl, nokia, nposl-3.0, ntp, oclc-2.0, odbl-1.0, ofl-1.1, ogtsl, osl-1.0, osl-2.0, osl-2.1, osl-3.0, php-3.0, postgresql, python-2.0, qpl-1.0, rpl-1.1, rpl-1.5, rpsl-1.0, rscpl, ruby, simpl-2.0, sissl, sleepycat, spl-1.0, unlicense, upl-1.0, vim, vsl-1.0, w3c, watcom-1.0, wtfpl, xnet, zlib, zpl-2.0
```

By that same logic, there are 83 potential additions:

```
aal, afl-1.1, afl-1.2, afl-2.0, afl-2.1, apache-1.1, apl-1.0, apsl-1.0, apsl-1.1, apsl-1.2, apsl-2.0, artistic-1.0, artistic-1.0-cl8, artistic-1.0-perl, bsl-1.0, catosl-1.1, cc-by-4.0, cc-by-sa-4.0, cddl-1.0, cecill-2.1, cnri-python, cpal-1.0, cpl-1.0, cua-opl-1.0, ecl-1.0, ecl-2.0, efl-1.0, efl-2.0, entessa, eudatagrid, eupl-1.1, fair, frameworx-1.0, hpnd, ijg, imatix, intel, ipa, ipl-1.0, lgpl-2.0, lpl-1.0, lpl-1.02, lppl-1.3c, miros, motosoto, mpl-1.0, mpl-1.1, mpl-2.0-no-copyleft-exception, multics, nasa-1.3, naumen, ncsa, ngpl, nokia, nposl-3.0, ntp, oclc-2.0, odbl-1.0, ogtsl, osl-1.0, osl-2.0, osl-2.1, php-3.0, postgresql, python-2.0, qpl-1.0, rpl-1.1, rpl-1.5, rpsl-1.0, rscpl, ruby, simpl-2.0, sissl, sleepycat, spl-1.0, upl-1.0, vim, vsl-1.0, w3c, watcom-1.0, xnet, zlib, zpl-2.0
```

File: spec/license_spec.rb [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
+    end
+
+    it "uses its SPDX name" do
+      spdx = find_spdx(id)
+      expect(spdx[1]["name"].gsub(/ only$/,"")).to eql(license["title"])
+    end
+
+    context "industry approval" do
+
+      # FSF approved the clear BSD, but doesn't use it's SPDX ID or Name
```
Comment (waldyrious) [inline spec/license_spec.rb]: Minor typo: "it's" --> "its". Also, shouldn't "Clear" be capitalized?

File: README.md [hunk 2]
```diff
@@ -8,11 +8,11 @@ Like a Choose Your Own Adventure site, but only much less interesting.
 
 A lot of repositories on GitHub.com don't have a license. GitHub provides a license chooser, but if you don't know anything about licenses, how are you supposed to make an informed decision?
 
-[ChooseALicense.com](http://www.choosealicense.com "Choose A Licence website") is designed to help people make an informed decision about licenses.
+[ChooseALicense.com](http://www.choosealicense.com "Choose A Licence website") is designed to help people make an informed decision about licenses by demystifying license choices through non-judgmental guidance.
 
 ## Immediate Goals
 
-* Non-partisan. Our goal is to help you find a license that meets *your* goals.
+* Non-judgemental. Our goal is to help you find a license that meets *your* goals.
```
Comment (mlinksva) [inline README.md]: judgmental is more common spelling

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -6,17 +6,20 @@ We love Pull Requests! Your contributions help make ChooseALicense.com great.
 
 So you want to contribute to ChooseALicense. Great! We welcome any help we can
 get. But first, please make sure you understand what
-[this site is all about](http://choosealicense.com/about).
+[this site is all about](http://choosealicense.com/about). It’s not a comprehensive list of all possible licenses.
 
-It’s not a comprehensive list of all possible licenses.
+## Adding new licenses
```
Comment (mlinksva) [inline CONTRIBUTING.md]: "new" could be left out

File: spec/license_spec.rb [hunk 4]
```diff
@@ -1,35 +1,32 @@
 require 'spec_helper'
 
-# List of licenses that need not be OSI approved
-LICENSE_WHITELIST = %w[
-  unlicense
-  cc0-1.0
```
Comment (mlinksva) [inline spec/license_spec.rb:6]: does this need to be whitelisted? Should be found by od_approved_licenses

File: spec/license_spec.rb [hunk 5]
```diff
@@ -1,35 +1,32 @@
 require 'spec_helper'
 
-# List of licenses that need not be OSI approved
-LICENSE_WHITELIST = %w[
-  unlicense
```
Comment (mlinksva) [inline spec/license_spec.rb:5]: Should be found by fsf_approved_licenses

File: spec/license_spec.rb [hunk 6]
```diff
@@ -1,35 +1,32 @@
...
-  cc0-1.0
-  wtfpl
```
Comment (mlinksva) [inline spec/license_spec.rb:7]: Should be found by fsf_approved_licenses

File: spec/license_spec.rb [hunk 7]
```diff
@@ -1,35 +1,32 @@
...
+
+    it "uses its SPDX name" do
+      spdx = find_spdx(id)
+      expect(spdx[1]["name"].gsub(/ only$/,"")).to eql(license["title"])
+    end
+
+    context "industry approval" do
+
+      # FSF approved the clear BSD, but doesn't use it's SPDX ID or Name
+      if license["id"] == "bsd-3-clause-clear"
```
Comment (mlinksva) [inline spec/license_spec.rb]: Maybe this should be in fsf_approved_licenses ?

File: spec/license_spec.rb [hunk 8]
```diff
@@ -1,35 +1,32 @@
 require 'spec_helper'
 
-# List of licenses that need not be OSI approved
-LICENSE_WHITELIST = %w[
-  unlicense
-  cc0-1.0
-  wtfpl
-  bsd-3-clause-clear
```
Comment (mlinksva) [inline spec/license_spec.rb:8]: This is also handled explicitly below. Not sure why the whitelist is needed at all.

**Comment by mlinksva** [comment]
Looks good overall! See line notes for minor comments.

File: spec/license_spec.rb (hunk 8 above)
Comment (benbalter) [inline spec/license_spec.rb:8]: The whitelist was part of the old test (which only looked at OSI approval), and is being removed by this pull request.

Comment (mlinksva) [inline spec/license_spec.rb:8]: Yikes. How did I fail to see that.

File: spec/license_spec.rb (hunk 7 above)
Comment (benbalter) [inline spec/license_spec.rb]: Good call. Moved via c801255.

**Comment by benbalter** [comment]
Thanks for the feedback @mlinksva. Believe I implemented all your suggestions, but glad to continue to improve if you have any more / I missed any / things change once we start using it. :smile:

**Pull request merged**